src/main/java/york/pharmacy/orders/OrderController.java

```diff
@@ -0,0 +1,61 @@
+package york.pharmacy.orders;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import jakarta.validation.Valid;
+import york.pharmacy.orders.dto.OrderRequest;
+import york.pharmacy.orders.dto.OrderResponse;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/orders")
+@RequiredArgsConstructor
+public class OrderController {
+
+    private final OrderService orderService;
+
+    // Create a new order
+    @PostMapping
+    public ResponseEntity<OrderResponse> createOrder(@Valid @RequestBody OrderRequest orderRequest) {
+        OrderResponse orderResponse = orderService.createOrder(orderRequest);
+        return new ResponseEntity<>(orderResponse, HttpStatus.CREATED);
+    }
+
+    // Create multiple orders (batch)
+    @PostMapping("/batch")
+    public ResponseEntity<List<OrderResponse>> batchCreateOrders(@Valid @RequestBody List<OrderRequest> orderRequests) {
+        List<OrderResponse> orderResponses = orderService.batchCreateOrders(orderRequests);
+        return new ResponseEntity<>(orderResponses, HttpStatus.CREATED);
+    }
+
+    // Get all orders
+    @GetMapping
+    public ResponseEntity<List<OrderResponse>> getAllOrders() {
+        List<OrderResponse> orders = orderService.getAllOrders();
+        return new ResponseEntity<>(orders, HttpStatus.OK);
+    }
+
+    // Get an order by its ID
+    @GetMapping("/{id}")
+    public ResponseEntity<OrderResponse> getOrderById(@PathVariable Long id) {
+        OrderResponse orderResponse = orderService.getOrderById(id);
+        return new ResponseEntity<>(orderResponse, HttpStatus.OK);
+    }
+
+    // Update an existing order by its ID
+    @PutMapping("/{id}")
+    public ResponseEntity<OrderResponse> updateOrder(@PathVariable Long id, @Valid @RequestBody OrderRequest orderRequest) {
+        OrderResponse orderResponse = orderService.updateOrder(id, orderRequest);
+        return new ResponseEntity<>(orderResponse, HttpStatus.OK);
+    }
+
+    // Delete an order by its ID
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Void> deleteOrder(@PathVariable Long id) {
+        orderService.deleteOrder(id);
```

Comment:
The `deleteOrder` service method throws an exception if the id doesn't exist, but it is not handled by the controller here. Something like
```
try {
    ...
} catch (Exception e) {
    return ResponseEntity.badRequest().build();
}
```
Would handle it. In fact, it might be worth using this pattern for all controller methods.